### PR TITLE
e2e, alerts, Add monitoring lane and tests

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR adds a monitoring lane to CNAO.
It also adds e2e tests to check CnaoDown alert

a followup PR is planned to introduce the tests using https.

**Special notes for your reviewer**:
PR relies on https://github.com/kubevirt/cluster-network-addons-operator/pull/1258 (first commit)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: automation/check-patch.e2e-monitoring-k8s.sh
Comment (qinqon) [inline automation/check-patch.e2e-monitoring-k8s.sh:17]: We are at 1.23 at main now.

File: test/e2e/monitoring/alerts_test.go
Comment (qinqon) [inline test/e2e/monitoring/alerts_test.go:52]: Do this with a Eventually.

File: test/e2e/monitoring/alerts_test.go
Comment (qinqon) [inline test/e2e/monitoring/alerts_test.go:70]:
```suggestion
			Expect(kubectl.KillPortForwardCommand(portForwardCmd)).ToSuccess()
```

File: test/e2e/monitoring/alerts_test.go
Comment (qinqon) [inline test/e2e/monitoring/alerts_test.go:52]: Do this with eventually

File: automation/check-patch.e2e-monitoring-k8s.sh
Comment (qinqon) [inline automation/check-patch.e2e-monitoring-k8s.sh:17]: Cnao is a k8s 1.23 at main branch

File: test/e2e/monitoring/prometheus_utils.go
Comment (qinqon) [inline test/e2e/monitoring/prometheus_utils.go:94]: Can this be a constant ?

**Review by qinqon — CHANGES_REQUESTED** [review]

File: test/e2e/monitoring/prometheus_utils.go
Comment (oshoval) [inline test/e2e/monitoring/prometheus_utils.go:51]: this wont work
we are using https, please use Bearer
see please 
https://github.com/kubevirt/cluster-network-addons-operator/pull/1144/files#diff-27325d74042ddc4a9b74aa4001b272ec23f03d9cd8c52a629e04bd93eb5f9490R710

Comment (RamLavi) [reply test/e2e/monitoring/prometheus_utils.go:51]: we do use a brearer, but you are right we need to use https

**Comment by RamLavi** [comment]
/holdapi
I'm having trouble seeing the rules in prometheus API.. CRC looks ok but doing it with k8s seems to be missing a piece.. @erkanerol do you have any idea perhaps?

File: test/e2e/monitoring/prometheus_utils.go
Comment (RamLavi) [reply test/e2e/monitoring/prometheus_utils.go:51]: after checking, it looks like http is the only working way. not sure why though..

**Comment by RamLavi** [comment]
The plan is to add another lane in project-infra after this PR is merged. 
This is the test output when running locally:
```
$ make test/e2e/monitoring
GOBIN=$(pwd)/build/_output/bin/ /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/build/_output/bin//go//bin//go install ./vendor/github.com/onsi/ginkgo/ginkgo
GINKGO=build/_output/bin/ginkgo GO=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/build/_output/bin//go//bin//go TEST_SUITE=test/e2e/monitoring TEST_ARGS="-test.v -test.timeout 3h -ginkgo.v" ./hack/functest.sh
+ source ./cluster/cluster.sh
++ export KUBEVIRT_PROVIDER=k8s-1.23
++ KUBEVIRT_PROVIDER=k8s-1.23
++ export KUBEVIRTCI_TAG=2203021427-243853c
++ KUBEVIRTCI_TAG=2203021427-243853c
++ KUBEVIRTCI_REPO=https://github.com/kubevirt/kubevirtci.git
++ CLUSTER_PATH=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci/
+ export OPERATOR_NAMESPACE=cluster-network-addons
+ OPERATOR_NAMESPACE=cluster-network-addons
++ cluster::kubeconfig
++ '[' k8s-1.23 '!=' external ']'
++ echo -n /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci//_ci-configs/k8s-1.23/.kubeconfig
+ export KUBECONFIG=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci//_ci-configs/k8s-1.23/.kubeconfig
+ KUBECONFIG=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci//_ci-configs/k8s-1.23/.kubeconfig
+ export KUBECTL=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci//cluster-up/kubectl.sh
+ KUBECTL=/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/_kubevirtci//cluster-up/kubectl.sh
+ build/_output/bin/ginkgo ./test/e2e/monitoring ../ -- -test.v -test.timeout 3h -ginkgo.v
=== RUN   TestE2E
Running Suite: Monitoring E2E Test Suite
========================================
Random Seed: 1646911459
Will run 1 of 1 specs

Prometheus Alerts when networkaddonsconfig CR is deployed and cluster-network-addons-operator deployment has no ready replicas 
  should issue CnaoDown alert
  /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/test/e2e/monitoring/alerts_test.go:48
STEP: Issuing a port forwarding command to access prometheus API
STEP: Delpoying CNAO CR with at least one component
STEP: Applying NetworkAddonsConfig spec:
macvtapcni: {}
STEP: Checking that condition "Available" status is set to True
STEP: setting CNAO operator deployment replicas to 0
STEP: waiting for the amount of time it takes the alert to fire
STEP: checking existence of alert
STEP: Removing NetworkAddonsConfig
STEP: Performing cleanup

• [SLOW TEST:304.428 seconds]
Prometheus Alerts
/home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/test/e2e/monitoring/alerts_test.go:17
  when networkaddonsconfig CR is deployed
  /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/test/e2e/monitoring/alerts_test.go:19
    and cluster-network-addons-operator deployment has no ready replicas
    /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/test/e2e/monitoring/alerts_test.go:42
      should issue CnaoDown alert
      /home/ralavi/go/src/github.com/kubevirt/cluster-network-addons-operator/test/e2e/monitoring/alerts_test.go:48
------------------------------

Ran 1 of 1 Specs in 307.149 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestE2E (307.16s)
PASS
You're using deprecated Ginkgo functionality:
=============================================
Ginkgo 2.0 is under active development and will introduce (a small number of) breaking changes.
To learn more, view the migration guide at https://github.com/onsi/ginkgo/blob/v2/docs/MIGRATING_TO_V2.md
To comment, chime in at https://github.com/onsi/ginkgo/issues/711

  You are using a custom reporter.  Support for custom reporters will likely be removed in V2.  Most users were using them to generate junit or teamcity reports and this functionality will be merged into the core reporter.  In addition, Ginkgo 2.0 will support emitting a JSON-formatted report that users can then manipulate to generate custom reports.

  If this change will be impactful to you please leave a comment on https://github.com/onsi/ginkgo/issues/711
  Learn more at: https://github.com/onsi/ginkgo/blob/v2/docs/MIGRATING_TO_V2.md#removed-custom-reporters

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=1.16.4


Ginkgo ran 1 suite in 5m13.602696057s
Test Suite Passed
```

File: test/e2e/monitoring/prometheus_utils.go
Comment (oshoval) [reply test/e2e/monitoring/prometheus_utils.go:51]: from the CNAO http will work because it still runs there, kube-rbac-proxy reads it
for https, you need the bearer and to change the link as in the above comments,

with CRC it works, for k8s vanilla Erkan said it doesn't work maybe because missing RBACs
because the namespace that Prometheus adds rbac to is `openshift-monitoring`, and we have namespace `monitoring` on U/S

File: test/e2e/monitoring/alerts_test.go
Comment (RamLavi) [reply test/e2e/monitoring/alerts_test.go:69]: Done

File: test/e2e/monitoring/alerts_test.go
Comment (RamLavi) [reply test/e2e/monitoring/alerts_test.go:52]: `checkForAlert` has an encapsulated `Eventually` so no need here

File: automation/check-patch.e2e-monitoring-k8s.sh
Comment (RamLavi) [reply automation/check-patch.e2e-monitoring-k8s.sh]: Done

**Comment by oshoval** [comment]
Not related to this PR:
worth to update To ginkgo 2.0 in a follow PR according the comment above

File: test/e2e/monitoring/prometheus_utils.go
Comment (RamLavi) [reply test/e2e/monitoring/prometheus_utils.go:94]: sure

**Comment by RamLavi** [comment]
/hold cancel
PR now tested and working locally

File: test/e2e/monitoring/prometheus_utils.go
Comment (RamLavi) [reply test/e2e/monitoring/prometheus_utils.go:53]: >from the CNAO http will work because it still runs there, kube-rbac-proxy reads it
for https, you need the bearer and to change the link as in the above comments,
the https works in that link,

We are not accessing CNAO here like former tests, we are directly querying prometheus API. The reason we don't need https is because we are wrapping the test with a [port-forward command](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-2915e6ecb9d8edd039184f1edf98c641cabb97148aafac11e59281f37c2b1ab3R23) to the monitoring namespace

>with CRC it works, for k8s vanilla Erkan said it doesn't work maybe because missing RBACs
because the namespace that Prometheus adds rbac to is openshift-monitoring, and we have namespace monitoring on U/S

The reason it worked on CRC is unrelated (the issue was a label selector in the prometheus CRD, taken cared of [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-6c6dc8ec613a709e2ccbaed045bac1a89bd91f5d60e8f0c303b98671931cf768R33)).

Comment (oshoval) [reply test/e2e/monitoring/prometheus_utils.go:53]: > We are not accessing CNAO here like former tests, we are directly querying prometheus API. The reason we don't need https is because we are wrapping the test with a [port-forward command](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-2915e6ecb9d8edd039184f1edf98c641cabb97148aafac11e59281f37c2b1ab3R23) to the monitoring namespace

In case it port forward the internal http port, then it doesn't fully test e2e imho, better to consider trying to improve it if its the case, on a follow PR please

Comment (RamLavi) [reply test/e2e/monitoring/prometheus_utils.go:53]: sure. I don't see why not do it on a separate PR

**Pull request merged**
